strawberryfields/tdm/tdmprogram.py

```diff
@@ -572,6 +576,38 @@ def unroll(self, shots):
             self.circuit = self.unrolled_circuit * shots
             return self
 
+        if self._is_space_unrolled:
+            raise ValueError("Program is space-unrolled and must be rolled before unrolling")
```

Comment:
I would expect that this error message will seem confusing to a user. Maybe naming the methods to call could help? (E.g., "Program is space-unrolled and cannot be unrolled. Call roll() so that it can be unrolled" or similar).